modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/order/test/CommerceOrderTest.java

```diff
@@ -841,6 +843,75 @@ public void testGetPlacedCommerceOrder() throws Exception {
 		_commerceAddressLocalService.deleteCommerceAddress(commerceAddress);
 	}
 
+	@Test
+	public void testValidateAccountOrdersLimit() throws Exception {
+		frutillaRule.scenario(
+			"Try to add 3 orders with account cart number limit set to 3"
+		).given(
+			"A B2B Site"
+		).and(
+			"A Group"
+		).and(
+			"A User"
+		).when(
+			"I try to get pending orders by channelGroupId and the accountId"
+		).then(
+			"I should have only 2 order"
+		);
+
+		Settings settings = _settingsFactory.getSettings(
+			new GroupServiceSettingsLocator(
+				_commerceChannel.getGroupId(),
+				CommerceConstants.ORDER_FIELDS_SERVICE_NAME));
+
+		ModifiableSettings modifiableSettings =
+			settings.getModifiableSettings();
+
+		modifiableSettings.setValue("accountCartMaxAllowed", "2");
+
+		modifiableSettings.store();
+
+		CommerceAccount commerceAccount =
+			_commerceAccountLocalService.addBusinessCommerceAccount(
+				"Test Business Account", 0, null, null, true, null,
+				new long[] {_user.getUserId()},
+				new String[] {_user.getEmailAddress()}, _serviceContext);
+
+		long commerceChannelGroupId = _commerceChannel.getGroupId();
+
+		_commerceOrderLocalService.addCommerceOrder(
+			_user.getUserId(), commerceChannelGroupId,
+			commerceAccount.getCommerceAccountId(),
+			_commerceCurrency.getCommerceCurrencyId());
+
+		_commerceOrderLocalService.addCommerceOrder(
+			_user.getUserId(), commerceChannelGroupId,
+			commerceAccount.getCommerceAccountId(),
+			_commerceCurrency.getCommerceCurrencyId());
+
+		try {
+			_commerceOrderLocalService.addCommerceOrder(
+				_user.getUserId(), commerceChannelGroupId,
+				commerceAccount.getCommerceAccountId(),
+				_commerceCurrency.getCommerceCurrencyId());
+		}
+		catch (CommerceOrderAccountLimitException
+					commerceOrderAccountLimitException) {
+
+			Assert.assertNotNull(commerceOrderAccountLimitException);
+		}
+
+		int pendingCommerceOrdersCount =
+			_commerceOrderService.getPendingCommerceOrdersCount(
+				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
+				StringPool.BLANK);
+
+		Assert.assertEquals(2, pendingCommerceOrdersCount);
+
+		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
```

Comment:
@andrea-ale-sbarra this logic should be in the tire down

Reply:
Hi @marco-leo I moved the logic of deleting orders to tear down method. I'm not sure if/how move also the logic about account, since no other methods seems to do this in the tear down